### PR TITLE
Bagels

### PR DESCRIPTION
**Pull request by DrEnzyme**
## About the PR

Bagels are made by using a rolling pin on a dough slice to make a dough rope, then cooking the dough rope in a microwave for 5 seconds. There are two types: a normal bagel and a poppyseed bagel. The poppyseed bagel requires a poppy and a dough rope and has a small (5u) quantity of Bicaridine inside of it in addition to its nutriment.

## Why / Balance

Made as an intro into the SS14 codebase to test feature creation from start to finish. I really wanted to add more options for the chef to use the ingredients that botany provides, and poppies don't currently have one outside of the advanced meds. Poppyseed bagels add a simple and relatively easy recipe for poppies to be used in, and are fairly underpowered (5u of Bicaridine compared to poppies 20u (!))

I'm aware that the recipe might be changed/killed or the PR declined pending a cooking rework.

## Technical details

No code changed. Three new prototypes added:
* Dough Rope
* Bagel
* Poppyseed Bagel

Recipes/Constructions created for Dough Rope (Rolling Pin on a Dough Slice), Bagel (Dough Rope, 5 secs in Microwave) and Poppyseed Bagel (Dough Rope, Poppy, 5 secs in Microwave)

New graphics have been added for bagels (bagel.rsi) and a dough rope has been added to the ingredients.rsi

## Media

Bagel
PoppyBagel

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

N/A

**Changelog**

:cl:
- add: Add bagels and poppy seed bagels.

### CODE REVIEW

**Comment by DrEnzyme** [comment]
I'm not sure why the tests are failing, does anyone have ideas?

**Comment by metalgearsloth** [comment]
Looks like your construction graph is invalid.

**Comment by Kadeo64** [comment]
bagels on bagelstation

**Comment by DrEnzyme** [comment]
> Looks like your construction graph is invalid.

Thanks Sloth! I think I found it. FoodDough instead of FoodDoughSlice in the construction recipe. Will fix.

**Comment by DrEnzyme** [comment]
> bagels on bagelstation

I wanna put some in but mapping is beyond me. They appear in the random food spawner though. Also the YML is a bit confusing when you realize there are two bagel.yml's

**Comment by metalgearsloth** [comment]
@DrEnzyme you still working on this bud?

**Comment by DrEnzyme** [comment]
Hey mates sorry I've been extremely busy these past few weeks. I would say I'll get around to making Partmedia's changes in the next few days but it might be longer. I'll definitely change poppy seed -> poppyseed and have the recipe use a packet of seeds (great idea btw). I'm not using the Bagel tag for anything, I didn't really know what it was for but I thought "better safe than sorry" and put a tag in just in case other code was going to be using it.

**Comment by Emisse** [comment]
Closed due to feature freeze May 10th-June 14th. Comment to have it reopen after this.

**Comment by DrEnzyme** [comment]
Hi @Emisse, I've made the changes that were suggested, can I have the PR re-opened?

**Comment by deltanedas** [comment]
not june 15th yet :trollface:

**Comment by DrEnzyme** [comment]
Phew now we're there. Can I re-open this PR now?

File: Resources/Prototypes/tags.yml [hunk 1]
```diff
@@ -27,6 +27,9 @@
 - type: Tag
   id: ATVKeys
 
+- type: Tag
+  id: Bagel
```
Comment (Partmedia) [inline Resources/Prototypes/tags.yml:31]: What is this tag being used for?

**Review by Partmedia — COMMENTED** [review]
Thanks for your updates, I've left a few review comments.

Sorry it's taken so long to get your PR in. Thanks for your work and your persistence. If you're able to take a look at the minor nitpicks I brought up, great. If not, I'll handle them and merge this for you.

**Comment by Partmedia** [comment]
Actually, I have this currently checked out so I'm just going to make the small changes for you. Please keep that in mind for next time.

**Comment by Partmedia** [comment]
Thanks for your (first!) contribution!

**Pull request merged**